client/asset/btc/rpcclient.go

```diff
@@ -225,6 +248,28 @@ func (wc *rpcClient) reconfigure(cfg *asset.WalletConfig, currentAddress string)
 			}
 		}
 
+		chainInfo := new(getBlockchainInfoResult)
+		if err := call(wc.ctx, cl, methodGetBlockchainInfo, nil, chainInfo); err != nil {
+			return false, fmt.Errorf("%s: %w", methodGetBlockchainInfo, err)
+		}
+		if !chainOK(wc.cloneParams.Network, chainInfo.Chain) {
+			return false, errors.New("wrong net")
+		}
+
+		// Ensure we can get a new address and decode it.
+		if wc.addrFunc != nil { // with an addrFunc, cannot check since it likely uses the prior rpcClient
+			return true, nil
+		}
+		var addrStr string
+		if err := call(wc.ctx, cl, methodNewAddress, nil, &addrStr); err != nil {
+			wc.log.Warnf("Failed to retrieve address (%v). Full wallet reconnect may be required.", err)
+			return true, nil // force a restart so connect uses wc.externalAddress
+		}
+		if _, err = wc.decodeAddr(addrStr, wc.chainParams); err != nil {
+			wc.log.Warnf("Failed to decode address (%v). Full wallet reconnect may be required.", err)
+			return true, nil // force a restart so connect uses wc.externalAddress
+		}
+
```

Comment:
[This change](https://github.com/buck54321/dcrdex/compare/2302618...buck54321:dcrdex:reconfig-connect) should resolve #2330 too, if you can live with that funky atomic pattern.

Reply:
> if you can live with that funky atomic pattern.

Yeah, I think so.  I think there's a missing `return` in the `if passedChecks {` branch, but otherwise makes sense.
Although it might be a problem if there's stuff going on that uses the replaced client that is briefly pointing to a bad node.

Reply:
Yeah, I'm not so sure about that now.  There's also some unsafe assignments some fields of `rpcClient` that we consider safe in `connect` by convention (always `connect`/`Connect` before using it).

Possibly the following tweak to the current PR could resolve the need to remake the accounts for forcing full reconnect (new `rpcClient` instance):

```go
		if wc.connectFunc != nil || wc.addrFunc != nil {
			return true, nil // back to full reconnect
		}
```

Reply:
It seemed like we should be checking versions too.

Reply:
> There's also some unsafe assignments some fields of `rpcClient` that we consider safe in `connect` by convention (always `connect`/`Connect` before using it).

Like `rpcConfig` and `ctx`? Yeah. we definitely rely on our assumption about the user synchronizing connect/reconfig, but that's not new.

Reply:
Right.  Not new, but I think we have to stick with that convention.

Reply:
> It seemed like we should be checking versions too.

I see your point.  I'll factor out the meat of the checks from the `connect` method.

Reply:
I think I now have this as clear and foolproof as it can get.  If things get tricky when trying to handle changes to the RPC connection settings, it's best to fallback to full reconnect instead of "live" reconfigure.  I incorporated one aspect of your suggestion, which is to validate the dep address regardless of "active use" of the wallet and force a full wallet restart if it's not owned.  Also the presence of a `wc.connectFunc` will trigger that.